docs/development/matchmaker.adoc

```diff
@@ -0,0 +1,83 @@
+
+[source, csharp]
+----
+// Attach an event handler for when the user is matched with other users.
+client.OnMatchmakeMatched += (object source, NMatchmakeMatchedEventArgs args) => {
+  INMatchmakeMatched matched = args.Matched;
+  // The match token is used to join a multiplayer match.
+  Debug.LogInfoFormat ("Match token: '{0}'", matched.Token);
+  foreach (var presence in matched.Presence) {
+    Debug.LogInfoFormat ("User(id: '{0}', handle: '{1}')", presence.UserId, presence.Handle);
+  }
+};
+----
+
+Send a message to add the user to the matchmaker pool.
+
+[source, csharp]
+----
+// Look for two opponents to be matched against.
+var message = NMatchmakeAddMessage.Default(2);
+client.Send (message, (INMatchmakeTicket ticket) => {
+  // Store the ticket so it can be used to cancel later.
+}, (INError error) => {
+  Debug.LogErrorFormat ("Send error: code '{1}' with '{0}'.", error.Message, error.Code);
+});
+----
+
+The `INMatchmakeTicket` is useful in case you want to remove the user from the matchmaker pool at some later point. In some games a user can cancel their action to matchmake at some later point and remove themselves being matched with other users.
+
+== Matchmake Remove
+
+After a user has sent a message to add themselves to the matchmaker pool you'll receive a `INMatchmakeTicket` which can be used to cancel the action at some later point.
+
+[source, csharp]
+----
+// Ticket is an instance of `INMatchmakeTicket`.
+var message = NMatchmakeRemoveMessage.Default(ticket);
+client.Send (message, (bool done) => {
+  // The user is now removed from the matchmaker pool.
+}, (INError error) => {
+  Debug.LogErrorFormat ("Send error: code '{1}' with '{0}'.", error.Message, error.Code);
+});
+----
+
+The user is now removed from the matchmaker pool.
+
+NOTE: **Matchmake remove on disconnect**
+A user will automatically be removed from the matchmaker pool by the server when they disconnect.
+
+== Join a Match
+
+To join a match after the event handler has notified the user their criteria is met and they've been given opponents you can use the `INMatchToken`.
+
+[source, csharp]
+----
+client.OnMatchmakeMatched += (object source, NMatchmakeMatchedEventArgs args) => {
+  INMatchmakeMatched matched = args.Matched;
+  // The match token is used to join a multiplayer match.
+  var message = NMatchJoinMessage.Default(matched.Token);
+  client.Send (message, (INMatch match) => {
+    Debug.Log ("Successfully joined match.");
+  }, (INError error) => {
+    Debug.LogErrorFormat ("Send error: code '{1}' with '{0}'.", error.Message, error.Code);
+  });
+};
+----
+
+The token makes it easy to join a match without an explicit match create action. The token enables the server to know that these users wanted to join a match and is able to create a match dynamically for them.
```

Comment:
Add: `Tokens are short-lived and must be used to join a match as soon as the client is ready. Once a token expires it can no longer be used in match join operations.`